---
title: Analysis of proportion of human pathogens for different phylums from JGI IMG
  database
author: "Miroshnikova Anastasia"
date: "8 August 2017"
output:
  html_document:
    css: 303_finaltask.css
    toc: yes
    toc_float: yes
  pdf_document:
    keep_tex: yes
    latex_engine: xelatex
    toc: yes
params: null
bibliography: srep39194.bib
urlcolor: blue
---

```{r setup, include=FALSE}
library(dplyr)
library(tidyr)
library(stringr)
library(ggplot2)
library(ggthemes)
library(lazyeval)
library(ggvis)
options(scipen = 5, digits = 3)
knitr::opts_chunk$set(echo=FALSE, message=FALSE, warning=FALSE, cache=TRUE)
```

## Introduction

The study to be published here is part of a student project which took place at [Bioinformatics Institute Summer School - 2017](http://bioinformaticsinstitute.ru/summer2017). The data set used in the study was downloaded from [JGI IMG database](https://img.jgi.doe.gov/cgi-bin/m/main.cgi)[^1] with multiple filters. The original study was based on @PaPrBaG.

[^1]: Here is the [direct link](https://raw.githubusercontent.com/Miffka/RAnalysis2FinalTask/master/IMG_all-more%2Bpathog.xls) to data download.

```{r readdata}
allinfo <- read.delim("https://raw.githubusercontent.com/Miffka/RAnalysis2FinalTask/master/IMG_all-more%2Bpathog.xls", stringsAsFactors = F)
allinfo01 <- allinfo %>% 
  filter(nchar(Phenotype) > 0)
```

The original data set contains `r nrow(allinfo)` records about sequenced bacterial genomes, but only `r nrow(allinfo01)` possess information about organism phenotype. The analyzed bacteria were presented by `r length(unique(allinfo01$Species))` species, which belonged to one of `r length(unique(allinfo01$Family))` families. These families belonged to one of theese phylums:

```{r phylums, tidy=TRUE}
unique(allinfo01$Phylum)
```

## Hypothesis

We were interested in obtaining reliable and self-consistent sequencing data, so we decided to find out whether there is a statistically significant difference in proportion of sequenced pathogens in different phylums of bacteria. For these purpose we used **R** (version `r getRversion()`) and two appropriate statistical tests called _Pearson's Chi-square_: $$X^2 = \sum_{}{}\frac{(O - E)^2}{E} $$
and _Fisher's exact test_: $$P_{cutoff} = \frac{(R_1!R_2!...R_m!)(C_1!C_2!...C_n!)}{N!\prod_{ij}^{}a_{ij}} $$

Our zero hypothesis is that proportion of sequenced pathogens over total sequenced samples of the phylum is the same for all phylums.

The workflow is as follows:

1. Read file
2. Prepare data for analysis
3. Run chi-square test and Fisher's exact test for all appropriate phylums and obtain p-value
4. Support the evidence with 
    + a table of results;
    + a plot that could help demonstrate them.

## Analysis

Because of database format, there are `r length(table(allinfo01$Phenotype))` different types of phenotype. That's why before the analysis we added a new variable with only two phenotype levels - either "Pathogen" or "Non-Pathogen".

```{r pre-analysis1}
allpathogens <- c(grep("^Pathogen", unique(allinfo01$Phenotype), value = TRUE), 
                  grep("[[:space:]]Pathogen", unique(allinfo01$Phenotype), value = TRUE))
allinfo02 <- allinfo01 %>% 
  mutate_(Phenotype.New = interp(~ifelse(var %in% allpathogens, "Pathogen", "Non-Pathogen"),
                                 var = as.name("Phenotype")))

```
```{r table1_new.phenotypes}
table01 <- table(allinfo02$Phylum, allinfo02$Phenotype.New)
```
```{r table2_corrected_table1}
table02 <- table01[apply(table01, 1, sum) > 8,]
```
It turned out that for some phylums there was not enough observations even for Fisher's exact test (less than 8 for whole phylum). These phylums, i.e. `r rownames(table01[apply(table01, 1, sum)<8,])` were excluded from the analysis.
For the other phylums there was no data on sequenced non-pathogens, so they were also excluded from analysis (`r rownames(table02[!apply(table02, 1, all) != 0,])`). The final data to be analyzed by Fisher's exact test were as follows:

```{r table4_and_table5}
(table04 <- table02[apply(table02, 1, all) != 0,])
table05 <- table04[apply(table04, 1, function(x){all(x > 5)}),]
```

For chi-sqared test the phylum `r rownames(table04)[!apply(table04, 1, function(x){all(x > 5)})]` was excluded.

So structure of our data was as follows:

```{r primary_graphics, fig.width=6, fig.height=5, fig.cap="Relative distribution of phylums in pathogenic and non-pahtogenic sequenced samples"}
allinfo03 <- allinfo02 %>% 
  filter(Phylum %in% rownames(table01[apply(table01, 1, sum) > 43,]))
ggplot(allinfo03, aes(x = "", fill = Phenotype.New))+
  geom_histogram(width = 1, position = "fill", stat = "count")+
  facet_wrap(~ Phylum, nrow = 1)+
  coord_polar(theta = "y")+
  scale_x_discrete(name = "", labels = c("", "", "", ""), breaks = NULL)+
  scale_y_continuous(name = "", breaks = NULL)+
  scale_fill_brewer(type = "qual", palette = 3, name = "Phenotype: ")+
  theme(strip.background = element_rect(fill = "blue"), 
        panel.background = element_rect(fill = "white"), 
        panel.spacing = unit(x = 0.1, units = "cm"),
        axis.title = element_text(size = 18, vjust = 1),
        legend.position = "bottom",
        legend.title = element_text(size = 16, vjust = 0),
        legend.text = element_text(size = 12),
        strip.text.x = element_text(size = 14, colour = "white", face = "bold"))
```

```{r analysis}
fisherres_t04 <- fisher.test(table04, workspace = 2e+06)
chisqres_t05 <- chisq.test(table05)
fisherres_t05 <- fisher.test(table05, workspace = 2e+06)
```

The p-value for the tests were $`r fisherres_t04$p.value`$ for Fisher's exact test and $`r chisqres_t05$p.value`$ for chi-squared test ($`r fisherres_t05$p.value`$ for the same data). This let us to say that proportion of pathogens is different for at least one phylum. 

For chi-squared test a matrix of residuals was
```{r visualization1}
chisqres_t05$residuals
```

```{r vizualization2, fig.width=10, fig.height=5, fig.cap="Pearson's residuals for different phylums", fig.subcap="The number of observations for each phylum is shown in log scale."}
chisqres_dt05 <- as.data.frame(chisqres_t05$residuals)
chisqres_dt05 <- chisqres_dt05 %>% 
  mutate(Freq2 = factor(sapply(Freq, function(x){
    if (x < -4){-4
    } else if (x < -2) {-2
    } else if (x < 0) {-1
    } else if (x > 4) {4
    } else if (x > 2) {2
    } else if (x > 0) {1
    }
  })))
chisqres_dt05$Count <- as.data.frame(chisqres_t05$observed)$Freq

ggplot(chisqres_dt05, aes(x = Var2, y = Count, fill = Freq2))+
  geom_col(position = "fill", width = 1)+
  facet_grid(Var1 ~ Var2, switch = "y", scales = "free_x", space = "free")+
  scale_x_discrete(name = "Phenotype", labels = c("", "", "", ""), breaks = NULL, position = "top")+
  scale_y_continuous(name = "Phylum", breaks = NULL)+
  scale_fill_brewer(type = "div", 
                    palette = 7, 
                    name = "Pearson's\nresiduals",
                    labels = c("<-4", "-4:-2", "-2:0", "0:2", "2:4", ">4"))+
  theme(strip.background = element_rect(fill = "white"), 
        panel.background = element_rect(fill = "white"), 
        panel.spacing = unit(x = 0.05, units = "cm"),
        axis.title = element_text(size = 18),
        legend.title = element_text(size = 16),
        legend.text = element_text(size = 12),
        strip.text = element_text(size = 14, face = "italic"),
        strip.text.y = element_text(angle = 180, hjust = 1, vjust = 0.5))
```

As we see, the proportion of sequenced non-pathogenic members of Actinobacteria and Bacteroidetes phylums is far more than expected, and proportion of non-pathogenic members of Firmicutes phylum is less than expected. 

##Conclusion

Our hypothesis of non-uniform distribution of sequenced pathogens over different phylums in the data base of interest was proven to be valid.

##Acknowledgements

Author is very grateful to the Organizing Commitee of __Bioinformatics Institute Summer School - 2017__  for letting a chance to obtain such interesting data.

## Bibliography
